addons/account/models/account_move.py

```diff
@@ -3729,6 +3729,18 @@ def name_get(self):
             result.append((line.id, name))
         return result
 
+    @api.model
+    def invalidate_cache(self, fnames=None, ids=None):
+        # Invalidate cache of related moves
+        if not fnames or 'move_id' in fnames:
+            f = self._fields['move_id']
+            line2move = self.env.cache._data.get(f)
+            if line2move:
+                move_ids = [move_id for line_id, move_id in line2move.items()
+                            if not ids or line_id in ids]
+                self.env['account.move'].browse(move_ids).invalidate_cache()
```

Comment:
This code breaks through the cache abstraction.  Use the cache API:
```suggestion
            field = self._fields['move_id']
            lines = self.env.cache.get_records(self, field) if ids is None else self.browse(ids)
            move_ids = {id_ for id_ in self.env.cache.get_values(lines, field) if id_}
            if move_ids:
                self.env['account.move'].invalidate_cache(ids=move_ids)
```